Bulldozer.FellowshipOne/Maps/Groups.cs

```diff
@@ -164,72 +181,129 @@ private void MapActivityMinistry( IQueryable<Row> tableData, long totalRows = 0
                 var activityName = row["Activity_Name"] as string;
                 var ministryActive = row["Ministry_Active"] as string;
                 var activityActive = row["Activity_Active"] as string;
+                var hasCheckin = row["Has_Checkin"] as bool?;
                 int? campusId = null;
+                var isServingMinistry = !string.IsNullOrWhiteSpace( ministryName ) && ministryName.RemoveSpaces().StartsWith( "SERV:", StringComparison.OrdinalIgnoreCase );
+                var isServingActivity = !string.IsNullOrWhiteSpace( activityName ) && activityName.RemoveSpaces().StartsWith( "SERV:", StringComparison.OrdinalIgnoreCase );
+                var activityHasPrefix = isServingActivity;
 
                 if ( ministryId.HasValue && !string.IsNullOrWhiteSpace( ministryName ) && !ministryName.Equals( "Delete", StringComparison.OrdinalIgnoreCase ) )
                 {
+                    var ministryServingGroupFK = "SERV_" + ministryId.Value.ToString();
+                    var activityServingGroupFK = "SERV_" + activityId.Value.ToString();
+                    var ministryServingGroupCascadeFK = "SERVT_" + ministryId.Value.ToString();
+                    var activityServingGroupCascadeFK = "SERVT_" + activityId.Value.ToString();
+
                     // check for a ministry group campus context
                     if ( ministryName.Any( n => ValidDelimiters.Contains( n ) ) )
                     {
                         campusId = campusId ?? GetCampusId( ministryName );
-                        if ( campusId.HasValue )
+                        if ( campusId.HasValue || isServingMinistry )
                         {
                             // strip the campus from the ministry name to use for grouptype (use the original name on groups though)
                             ministryName = StripPrefix( ministryName, campusId );
```

Comment:
`StripPrefix` does not handle if "SERV: <CAMPUS NAME>" pattern is followed. Which would be necessary for  serv to function as well because you were also using startswith "SERV:"